android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/RibEvents.kt

```diff
@@ -20,24 +20,96 @@ import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.rx2.asObservable
 
-public class RibEvents private constructor() {
+public object RibEvents {
 
-  private val _events = MutableSharedFlow<RibEvent>(0, 1, BufferOverflow.DROP_OLDEST)
-  public val events: Observable<RibEvent> = _events.asObservable()
+  private val mutableRouterEvents =
+    MutableSharedFlow<RibRouterEvent>(0, 1, BufferOverflow.DROP_OLDEST)
+  private val mutableRibDurationEvents =
+    MutableSharedFlow<RibActionInfo>(0, 1, BufferOverflow.DROP_OLDEST)
+
+  @JvmStatic
+  public val routerEvents: Observable<RibRouterEvent> = mutableRouterEvents.asObservable()
+
+  @JvmStatic
+  public val ribActionEvents: Observable<RibActionInfo> = mutableRibDurationEvents.asObservable()
 
   /**
    * @param eventType [RibEventType]
    * @param child [Router]
    * @param parent [Router] and null for the root ribs that are directly attached to
    *   RibActivity/Fragment
    */
-  public fun emitEvent(eventType: RibEventType, child: Router<*>, parent: Router<*>?) {
-    _events.tryEmit(RibEvent(eventType, child, parent))
+  public fun emitRouterEvent(eventType: RibEventType, child: Router<*>, parent: Router<*>?) {
+    mutableRouterEvents.tryEmit(RibRouterEvent(eventType, child, parent))
+  }
+
+  /**
+   * Calls related RIB action (e.g. didBecomeActive) and emits emission of ATTACHED/DETACHED events
+   * for each RIB component.
+   *
+   * @param ribAction The related RIB action type. e.g. didBecomeActive, willLoad, etc
+   * @param ribCallerClassType Related RIB component class type
+   * @param ribComponentType The RIB component type (e.g. Interactor, Router, Presenter, Worker)
+   * @param ribEventType RIB event type (e.g. ATTACH/DETACH)
+   */
+  internal inline fun <T : Any> triggerRibActionAndEmitEvents(
+    ribCallerClassType: T,
+    ribComponentType: RibComponentType,
+    ribEventType: RibEventType,
+    ribAction: () -> Unit,
+  ) {
+    val ribClassName = ribCallerClassType.asQualifiedName()
+    ribClassName?.emitRibEventAction(ribComponentType, ribEventType, RibActionState.STARTED)
```

Comment:
This function extension doesn't make sense. Just take a string class name as a param on the method.